package/MDAnalysis/coordinates/H5MD.py

```diff
@@ -0,0 +1,420 @@
+                                         with shape (n_atoms, 3)
+                        +-- units <str>
+                \-- (data)
+                    \-- (dt)
+                        \-- [step] <int>, gives frame number
+                        \-- [value] <float>, gives dt
+                    \-- (lambda)
+                        \-- [step] <int>, gives frame
+                        \-- [value] <float>, gives lambda
+                    \-- (step)
+                        \-- [step] <int>, gives frame
+                        \-- [value] <int>, gives step
+
+
+    .. note::
+        The reader does not currently read mass data.
+
+    """
+
+    format = 'H5MD'
+    # units are added from h5md file
+    units = {'time': None,
+             'length': None,
+             'velocity': None,
+             'force': None}
+    # Translate H5MD units (https://nongnu.org/h5md/modules/units.html)
+    # to MDAnalysis units.
+    _unit_translation = {
+        'time': {
+            'ps': 'ps',
+            'fs': 'fs',
+            'ns': 'ns',
+            'second': 'second',
+            'sec': 'sec',
+            's': 's',
+            'AKMA': 'AKMA',
+            },
+        'length': {
+            'Angstrom': 'Angstrom',
+            'angstrom': 'Angstrom',
+            'A': 'Angstrom',
+            'nm': 'nm',
+            'pm': 'pm',
+            'fm': 'fm',
+            },
+        'velocity': {
+            'Angstrom ps-1': 'Angstrom/ps',
+            'A ps-1': 'Angstrom/ps',
+            'Angstrom fs-1': 'Angstrom/fs',
+            'A fs-1': 'Angstrom/fs',
+            'Angstrom AKMA-1': 'Angstrom/AKMA',
+            'A AKMA-1': 'Angstrom/AKMA',
+            'nm ps-1': 'nm/ps',
+            'nm ns-1': 'nm/ns',
+            'pm ps-1': 'pm/ps',
+            'm s-1': 'm/s'
+            },
+        'force':  {
+            'kJ mol-1 Angstrom-1': 'kJ/(mol*Angstrom)',
+            'kJ mol-1 nm-1': 'kJ/(mol*nm)',
+            'Newton': 'Newton',
+            'N': 'N',
+            'J m-1': 'J/m',
+            'kcal mol-1 Angstrom-1': 'kcal/(mol*Angstrom)',
+            'kcal mol-1 A-1': 'kcal/(mol*Angstrom)'
+            }
+    }
+    _Timestep = Timestep
+
+    def __init__(self, filename, convert_units=True, **kwargs):
+        """
+        Parameters
+        ----------
+        filename : str or h5py.File
+            trajectory filename or open h5py file
+        convert_units : bool (optional)
+            convert units to MDAnalysis units
+        **kwargs : dict
+            General reader arguments.
+        """
+        if not HAS_H5PY:
+            raise RuntimeError("Please install h5py")
+        super(H5MDReader, self).__init__(filename, **kwargs)
+        self.filename = filename
+        self.open_trajectory()
+        self.has_positions = 'position' in self._particle_group
+        if not self.has_positions:
+            raise ValueError("'position' group must be in 'particles' group")
```

Comment:
Are positions required by the H5MD standard? Or are, say, velocity-only trajectories allowed? MDAnalysis would allow it.

Reply:
No, positions isn't required by the H5MD standard, but that's where the reader pulls `self.n_atoms` from, so I thought it would be simplest for the reader to require positions. 

I can change it to try pulling `n_atoms` from other possible groups, like have it look for `/position` first, then `/velocity`, etc. But in the case that there are _no_ groups in `/particles/group1` , there wouldn't be anywhere to pull `n_atoms` from unless it were defined in some other optional group, like `/particles/connectivity` maybe.